Author: Ben Hutchings <ben@decadent.org.uk>
Date: Sun, 29 Jan 2017 00:25:50 +0000
Description: Fix linker flags for test cases
 We need to explicitly disable use of a dynamic linker.  When using
 Debian's binutils 2.27-9, ld succeeds but appears to generate broken
 headers that cause the tests to fail.  When using binutils 2.28, ld
 fails, reporting "Not enough room for program headers, try linking
 with -N".
Bug-Debian: https://bugs.debian.org/842446

--- a/tests/Makefile.am
+++ b/tests/Makefile.am
@@ -29,7 +29,7 @@ test-i386.pecoff: test_lds = $(srcdir)/t
 test-i386.pecoff: ASFLAGS += -m32
 test-i386.pecoff: LDFLAGS += -m32
 
-test-%.elf: LDFLAGS = -nostdlib -T $(test_lds)
+test-%.elf: LDFLAGS = -nostdlib -T $(test_lds) -Wl,--no-dynamic-linker
 test-%.elf: test-%.$(OBJEXT) $(test_lds)
 	$(LINK) $<
 
